presto-main/src/main/java/com/facebook/presto/execution/scheduler/LegacySqlQueryScheduler.java

```diff
@@ -89,13 +106,25 @@
     private final SplitSchedulerStats schedulerStats;
 
     private final QueryStateMachine queryStateMachine;
-    private final SubPlan plan;
+    private final AtomicReference<SubPlan> plan = new AtomicReference<>();
     private final StreamingPlanSection sectionedPlan;
     private final StageId rootStageId;
     private final boolean summarizeTaskInfo;
     private final int maxConcurrentMaterializations;
 
-    private final Map<StageId, StageExecutionAndScheduler> stageExecutions;
+    // The following fields are required by adaptive optimization in runtime.
+    private final Session session;
+    private final FunctionManager functionManager;
+    private final List<PlanOptimizer> runtimePlanOptimizers;
+    private final WarningCollector warningCollector;
+    private final PlanNodeIdAllocator idAllocator;
+    private final PlanVariableAllocator variableAllocator;
+    private final SectionExecutionFactory sectionExecutionFactory;
+    private final RemoteTaskFactory remoteTaskFactory;
+    private final SplitSourceFactory splitSourceFactory;
+    private static final Logger log = Logger.get(LegacySqlQueryScheduler.class);
```

Comment:
nit: Usually we define `log` as a very first field in the class and separate it from the other fields